src/RadixDlt.NetworkGateway.Abstractions/Numerics/TokenAmount.cs

```diff
@@ -154,9 +152,9 @@ public static TokenAmount FromDecimalString(string decimalString)
 
     public static TokenAmount operator -(TokenAmount a, TokenAmount b) => (a.IsNaN() || b.IsNaN()) ? NaN : new TokenAmount(a._subUnits - b._subUnits);
 
-    public static TokenAmount operator *(TokenAmount a, TokenAmount b) => (a.IsNaN() || b.IsNaN()) ? NaN : new TokenAmount(a._subUnits * b._subUnits);
+    public static TokenAmount operator *(TokenAmount a, TokenAmount b) => (a.IsNaN() || b.IsNaN()) ? NaN : new TokenAmount(a._subUnits * b._subUnits / _divisor);
```

Comment:
I'd like us to wrap the multiplication in brackets to make clear that it is done first, and so that we avoid losing precision. Same below :)

Also, can we be consisted about whether we use `FromSubUnits` or `new TokenAmount`?